tb_plugin/torch_tb_profiler/profiler/trace.py

```diff
@@ -61,23 +69,51 @@ def __init__(self, data):
         # torch.profiler.profile.step will invoke record_function with name like "ProfilerStep#5"
         self.step = int(self.name.split("#")[1])
 
+class MemoryEvent(BaseEvent):
+    def __init__(self, type, data):
+        super().__init__(type, data)
+        self.scope = data.get("s", "")
 
-def create_event(event):
-    try:
-        type = event.get("ph")
-        if type != "X":
+    @property
+    def device_type(self):
+        dtype = self.args.get("Device Type")
+        if dtype is None:
+            return None
+
+        try:
+            return DeviceType(dtype)
```

Comment:
why not directly return the dtype?

Reply:
the dtype is integer and I want to convert to DeviceType enum